### DISCUSSION TITLE
I'm not getting completin candidates from nixd

### DISCUSSION THREAD

**Discussion by elzorrorebelde** [emacs-lsp/lsp-mode#4487] [Q&A] [answered]
I'm trying to get auto-completion in nix-mode but is impossible. [show-log.log](https://github.com/user-attachments/files/16118462/show-log.log)

I added a client:

```emacs-lisp
(after! lsp-mode
  (lsp-register-client
   (make-lsp-client :new-connection (lsp-stdio-connection "nixd")
                    :major-modes '(nix-mode)
                    :priority 1
                    :server-id 'nixd)))
```

Added the minimum amount of [server configuration](https://github.com/nix-community/nixd/blob/main/nixd/docs/configuration.md#configuration) to my flake.nix:

```nix
{ inputs, ... }:
{
  # NixOS configuration.
  nix.nixPath = [ "nixpkgs=${inputs.nixpkgs}" ];
}
```

but I get no completion candidates still.

I'm using this version of nixd:

```
➜  nixd --version
nixd, version: nightly
```

**Comment by jcs090218** [comment, marked as answer]
The log looks normal to me. 🤔 A Nix client is already specified in this project. See https://github.com/emacs-lsp/lsp-mode/blob/master/clients/lsp-nix.el.

I've tried it on macOS, and it works fine. It has only a few completions, but I can get it to work.
